docs/pre-commit.md

```diff
@@ -0,0 +1,33 @@
+# Using pre-commit locally
+
+### Install and setup pre-commit
+- Install:
+  - Using pip: `pip install pre-commit`
+  - Using Homebrew: `brew install pre-commit`
+
+- Setup
+  - run `pre-commit install` to set up the git hook script
+
+### Things to keep in mind regarding the git authoring lifecycle (HEAD, Staged, Unstaged) and pre-commit hooks
+
+Running hooks on unstaged changes can lead to both false-positives and false-negatives during committing. pre-commit only runs on the staged contents of files by temporarily saving the contents of your files at commit time and stashing the unstaged changes while running hooks.
+
+> pre-commit itself will never touch the staging area. These are good ways to silently break commits. In my mind this is one of the worst things that lint-staged does and suggests -- hooks are very frequently not perfect and magically changing what's being committed should not be taken lightly.
+
+Always stage the the changes you intend to commit before running `pre-commit`. If a hook changes our code and fails you will need to manually stage the changes made by `pre-commit`, viewing them one by one with `git add -p` or your preferred git GUI.
+
+### Running the pre-commit hook directly from the .git folder
+
+- run `bash .git/hooks/pre-commit` to execute pre-commit hook without committing
+
+### How to respond to failed pre-commit hooks
+
+- If when running `pre-commit` you change files that weren't intended to be changed, run `git add -p` to go through individual portions of changed code that you can either stage or remove.
+- Once finished staging changes you can run `git checkout -- .` which will revert all unstaged changes in the branch (nuclear option)
+
+### Write commit a message so it doesn't get blown away by a failed pre-commit hook
+
+- run `git commit` without a comment when that will then run pre-commit and should it fail you won't lose your commit message. Once it passes it will prompt you to input a commit message.
```

Comment:
```suggestion
- run `git commit` without a comment and that will then run pre-commit and should it fail you won't lose your commit message. Once it passes it will prompt you to input a commit message.
```